src/orient/rational-calc.orient

```diff
@@ -0,0 +1,354 @@
+    describe(expander_parents, "Number of Chung Expander parents of a node in the PoRep window graph")
+    describe(drg_layers, "Number of layers in the DRG")
+
+    describe(node_size, "Size of a node (bytes)")
+    describe(node_size_gb, "Size of a node (GiB)")
+    describe(kdf_hash_size, "Size of the input to the KDF hash (bytes)")
+    describe(kdf_hash_size_gb, "Size of the input to the KDF hash (GiB)")
+
+    sector_size = sector_size_gb * b_in_gb
+    graph_parents = drg_parents + expander_parents
+    nodes = sector_size / node_size 
+    node_size_gb = node_size / b_in_gb 
+    kdf_hash_size_gb = graph_parents * node_size_gb
+    kdf_hash_size = kdf_hash_size_gb * b_in_gb
+
+    window_size = window_size_mb * 1024 * 1024
+    windows = sector_size / window_size
+    window_nodes = nodes / windows
+
+  Time:
+    describe(porep_time, "Time to seal a sector (PoRep proof)", secs)
+    describe(porep_time_parallel, "Time to seal a sector (PoRep proof) with parallelization", secs)
+    describe(porep_proof_gen, "Time to generate a PoRep proof (SNARK, witness generation, loading data from storage medium)", secs)
+    describe(porep_proof_gen_parallel, "Time to generate a PoRep proof (SNARK, witness generation, loading data from storage medium) in parallel", secs)
+
+    porep_time = porep_proof_gen + porep_commit_time + encoding_time
+    porep_time_parallel = porep_proof_gen_parallel + porep_commit_time_parallel + encoding_time_parallel
+
+  Commitment:
+    describe(window_comm_tree_time, "Time to generate a merkle tree of stacked windows")
+    describe(window_comm_leaves_time, "Time to commit columns of a window")
+
+    describe(commit_size, "Size of a commitment (bytes)")
+    describe(commr_size, "Size of CommR (bytes)")
+    describe(commd_size, "Size of CommD (bytes)")
+    describe(commrlast_size, "Size of CommRLast (bytes)")
+
+    describe(commr_tree_depth, "Number of layers of the merkle tree with root hash CommR")
+
+    describe(seal_onchain_commitments_size, "Size of the seal commitments", bytes)
+    describe(porep_commit_time, "Time to generate CommC, CommR and CommQ", secs)
+    describe(porep_commit_time_mins, "Time to generate CommC, CommR and CommQ", mins)
+    describe(porep_commit_time_parallel, "Time to generate CommC, CommR and CommQ in parallel", secs)
+
+
+    describe(commr_time, "Time to generate CommR (seconds)")
+    describe(commd_time, "Time to generate CommD (seconds)")
+    describe(commc_time, "Time to generate CommC (seconds)")
+
+    commit_size = cid_size
+    // TODO: should be missing *2
+    window_comm_tree_time = merkle_tree_hash_time * window_nodes
+    window_comm_leaves_time = column_leaf_hash_time * windows
+    commr_size = commit_size
+    commd_size = commit_size
+    commrlast_size = commit_size
+
+    seal_onchain_commitments_size = commd_size + commr_size
+    // seal_commitment_time = commr_time
+
+    commd_time = merkle_tree_datahash_time * nodes
```

Comment:
Here as well, the number of nodes to be hashed in a tree should be `nodes *  2 - 1`.

Reply:
fixed in latest